Automattic/wp-calypso
# Customer Home: Add Mastering Gutenberg Card

#### Changes proposed in this Pull Request

This is very much a WIP; I still need to create the videos with some more detailed annotations/text, and select a thumbnail for them, along with making sure the support links are localised etc.

This adds another card with some Gutenberg resources.

#### Testing instructions

Copy and paste this into `client/my-sites/customer-home/locations/primary/index.jsx`

```jsx
/**
 * External dependencies
 */
import React from 'react';

/**
 * Internal dependencies
 */
import GoMobile from 'my-sites/customer-home/cards/features/go-mobile';
import ChecklistSiteSetup from 'my-sites/customer-home/cards/primary/checklist-site-setup';
import MasteringGutenberg from 'my-sites/customer-home/cards/education/mastering-gutenberg';
import QuickLinks from 'my-sites/customer-home/cards/primary/quick-links';

const cardComponents = {
	'home-feature-go-mobile-phones': GoMobile,
	'home-primary-checklist-site-setup': ChecklistSiteSetup,
	'home-primary-quick-links': QuickLinks,
	'home-education-mastering-gutenberg': MasteringGutenberg,
};

const Primary = ( { checklistMode, cards } ) => {
	// Always ensure we have primary content.
	if ( cards && cards.length < 1 ) {
		cards = [ 'home-primary-quick-links' ];
	}
	return (
		<div className="primary">
			{ cards &&
				cards.map(
					( card, index ) =>
						cardComponents[ card ] &&
						React.createElement( cardComponents[ card ], {
							key: index,
							checklistMode: card === 'home-primary-checklist-site-setup' ? checklistMode : null,
						} )
				) }
			<MasteringGutenberg />
		</div>
	);
};

export default Primary;

```

A list of links should appear. The card should not appear for sites unable to use Gutenberg.

<img width="722" alt="Screenshot 2020-04-07 at 22 31 11" src="https://user-images.githubusercontent.com/43215253/78721460-81644280-791f-11ea-9e82-0d510b209245.png">

Clicking one should open a modal with a video and a link to a support document on it. Verify that both of these work (the video should autoplay).

<img width="703" alt="Screenshot 2020-04-07 at 22 30 28" src="https://user-images.githubusercontent.com/43215253/78721435-78737100-791f-11ea-8d0a-7fbc767a0687.png">

The card should be dismissable, but hidden if a user cannot use Gutenberg (based on the `isClassicEditorForced` selector). 

Fixes #39132
